editor/plugins/org.fusesource.ide.project/src/org/fusesource/ide/project/camel/ui/CamelFacetInstallationPage.java

```diff
@@ -0,0 +1,191 @@
+		super( "camel.facet.install.page"); //$NON-NLS-1$
+		setTitle(Messages.NewCamelProject_FacetInstallationPage);
+		setDescription(Messages.NewCamelProject_FacetInstallationPageDesc);
+		
+	}
+
+	public void setConfig(Object config) {
+		System.out.println("config = " + config);
+		System.out.println("set config");
+		this.model = (IDataModel)config;
+		IDataModel o =(IDataModel) model.getProperty(FacetInstallDataModelProvider.MASTER_PROJECT_DM);
+		//FacetDataModelMap map = (FacetDataModelMap) model.getProperty(IFacetProjectCreationDataModelProperties.FACET_DM_MAP);
+		if( o == null ) {
+			// This is likely a pre-existing project having the camel facet added. We should avoid customizing project structure
+			model.setProperty(UPDATE_PROJECT_STRUCTURE, false);
+		}
+		if( o != null ) {
+			// This is a new camel project. We can customize the structure all we want. 
+			model.setProperty(UPDATE_PROJECT_STRUCTURE, true);
+			o.addListener(new IDataModelListener() {
+				public void propertyChanged(DataModelEvent event) {
+					System.out.println("Property changed");
+					if( IFacetProjectCreationDataModelProperties.FACET_ACTION_MAP.equals(event.getPropertyName())) {
+						updateWidgetsFromModel();
+					}
+				}
+			});
+		}
+	}
+
+	public void createControl(Composite parent) {
+		setControl(createTopLevelComposite(parent));
+	}
+	
+	private void changePageStatus(){
+		boolean webfound = hasWebFacet();
+		boolean shouldUpdateStructure = model.getBooleanProperty(UPDATE_PROJECT_STRUCTURE);
+		boolean displayContentFolder = webfound || shouldUpdateStructure;
+
+		if(displayContentFolder && !validFolderName(contentFolder.getText())){
+			setErrorMessage(Messages.NewCamelProject_FacetInstallationPage_ContentRootError);
+		} else {
+			setErrorMessage(null);
+		}
+		setPageComplete(isPageComplete());
+	}
+	
+	private boolean validFolderName(String folderName) {
+		IWorkspace ws = ResourcesPlugin.getWorkspace();
+		return ws.validateName(folderName, IResource.FOLDER).isOK();
+	}
+
+	
+	protected Composite createTopLevelComposite(Composite parent) {
+		final Composite composite = new Composite(parent, SWT.NONE);
+		composite.setLayout(new GridLayout(1, false));
+		createProjectGroup(composite);
+//		//set page status
+		changePageStatus();
+		
+		return composite;
+	}
+
+	private void createProjectGroup(Composite parent){
+		Composite prjGroup = new Composite(parent, SWT.NONE);
+		prjGroup.setLayout(new GridLayout(1, false));
+		prjGroup.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		
+		this.contentRootLabel = new Label(prjGroup, SWT.NONE);
+		this.contentRootLabel.setText(Messages.NewCamelProject_ContentRootLabel);
+		this.contentRootLabel.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+
+		this.contentFolder = new Text(prjGroup, SWT.BORDER);
+		this.contentFolder.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		this.contentFolder.setData("label", this.contentRootLabel); //$NON-NLS-1$ // wtf??
```

Comment:
wtf??
Where the "label" data attribute is used? preferable to use a constant in order to be able to find easily